pkg/compliance/agent/agent.go

```diff
@@ -92,19 +95,43 @@ func RunChecksFromFile(reporter event.Reporter, file string, options ...checks.B
 // Run starts the Compliance Agent
 func (a *Agent) Run() error {
 	a.scheduler.Run()
-	onCheck := func(check check.Check) error {
-		return a.scheduler.Enter(check)
+
+	defer status.Set(
+		"Checks",
+		expvar.Func(func() interface{} {
+			return a.builder.GetCheckStatus()
+		}),
+	)
+
+	onCheck := func(rule *compliance.Rule, check compliance.Check, err error) bool {
+		if err != nil {
+			log.Errorf("%s: check not scheduled: %v", rule.ID, err)
+			return true
+		}
+
+		err = a.scheduler.Enter(check)
+		if err != nil {
+			log.Errorf("%s: failed to schedule check: %v", rule.ID, err)
+			return false
+		}
+
+		return true
 	}
 	return a.buildChecks(onCheck)
 }
 
-func runCheck(check check.Check) error {
-	log.Infof("%s: Running check: %s [version=%s]", check.ID(), check.String(), check.Version())
-	err := check.Run()
+func runCheck(rule *compliance.Rule, check compliance.Check, err error) bool {
+	if err != nil {
+		log.Infof("%s: Not running check: %v", rule.ID, err)
+		return true
+	}
+
+	log.Infof("%s: Running check: %s [version=%s]", rule.ID, check.String(), check.Version())
+	err = check.Run()
 	if err != nil {
 		log.Errorf("%s: Check failed: %v", check.ID(), err)
```

Comment:
Sanity check: we don't want to return false here?

Reply:
Same here - we want to continue enumerating other checks.